src/coreclr/jit/ifconversion.cpp

```diff
@@ -558,10 +558,11 @@ bool OptIfConversionDsc::optIfConvert()
     }
 
     // Verify the test block ends with a condition that we can manipulate.
+    // Assuming that an AND connected to a JTRUE is always a valid chain.
```

Comment:
What constitutes a "valid chain"? What about e.g. `AND(x, y)` where x and y are locals or any arbitrary expressions?

Reply:
>What about e.g. AND(x, y) where x and y are locals or any arbitrary expressions?

Having a CCMP would make it explicit what is valid and what isn't